package-res/resources/web/pentaho/util/spec.js

```diff
@@ -0,0 +1,222 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "./object",
+  "../lang/OperationInvalidError"
+], function(O, OperationInvalidError) {
+
+  "use strict";
+
+  /**
+   * Map of merge operation name to operation handler function.
+   *
+   * @type {Object.<string, function>}
+   * @see mergeOne
+   */
+  var _mergeHandlers = {
+    "replace": mergeOperReplace,
+    "merge": mergeOperMerge,
+    "add": mergeOperAdd
+  };
+
+  /**
+   * @classDesc A singleton class that contains utilities related to configuration.
+   *
+   * @name pentaho.util.Spec
+   * @class
+   * @amd {pentaho.util.Spec} pentaho/util/spec
```

Comment:
I should've take a closer look at the code instead of just looking at the class description. 
Given that info I would leave the util where it is but remove any comments that refer configuration.